In [ ]:
import sys, os
os.chdir("../..")
import slot_extraction.utilities.slots as slt

from datasets import DatasetDict, load_dataset

dataset = load_dataset("multi_woz_v22")

filtered = slt.filter_dataset(dataset['test'])
slot_data = slt.construct_slot_extraction_data(filtered)

The following results where obtained using the Dataset ground truth and the same evaluation metrics as in the evaluation notebook, but removing the question mark slots as we considered they were part of the third task.

## Predictions results **with** similarity classification

In [2]:
slot_data['predictions'] = slot_data['utterance'].apply(slt.predict_ner)

slt.get_evaluation_metrics(slot_data)

Extrange dom_slot is:  with value:  star
Extrange dom_slot is:  with value: :30
Extrange dom_slot is:  with value:  oriental
Extrange dom_slot is:  with value:  hotel
Extrange dom_slot is:  with value:  guest house


(0.8734560708459567, 0.9099295945617868, 0.8913198573127228)

As it can be observed, the results obtained without taking into account the question mark slots is the following and using the similarity classifier for normalization are the following:

- Precision: 87.3%
- Recall: 90.9%
- F1-score: 89.1%

## Prediction results **without** similarity classification

In [3]:
slot_data['predictions2'] = slot_data['utterance'].apply(slt.predict_ner, args=(False,))
slt.get_evaluation_metrics(slot_data, predicition_column='predictions2')

(0.8179911442554183, 0.8521485797523671, 0.8347205707491081)

As it can be observed, the results obtained without taking into account the question mark slots is the following and **without** the similarity classifier for normalization are the following:

- Precision: 81.8%
- Recall: 85.2%
- F1-score: 83.5%